### PR TITLE
Feature/purchase: Added first version of purchase endpoint

### PR DESCRIPTION
**Pull request by ArturPimentel**
This new GraphQL endpoint serves two purposes: inform the billing options of an app and make its purchase possible.
This adds the following mutation to `vtex.billing`:
`purchase(appName: String!, termsOfUseAccepted: Boolean = false): PurchaseResponse`
Which returns the following type:
```
type PurchaseResponse {
  purchased: Boolean!
  billingOptions: String
}
```

In this type, we have:
- `purchased`: whether the purchase of the app was made
- `billingOptions`: JSON-formatted version of the app's billing options, made for processing by any other application

When used to get billing options, set the mutation parameter `termsOfUseAccepted` to false, e.g.: 
```
mutation{
  purchase(app:"vtexhackathon05.iago", termsOfUseAccepted: false) {
    purchased
    billingOptions
  }
}
```

The expected output should be
```
{"data":{"purchase":{"purchased":false,"billingOptions":"{\"deactivationRoute\":\"billingListeningRoute\",\"policies\":[{\"billing\":{\"items\":[{\"fixed\":100,\"itemCurrency\":\"BRL\"}],\"taxClassification\":\"software\"},\"currency\":\"USD\",\"plan\":\"default\"}],\"termsURL\":\"http://route.to.the/terms\",\"version\":\"0.1\"}"}}}
```

When used to purchase an app, set the mutation parameter `termsOfUseAccepted` to true, e.g.: 
```
mutation{
  purchase(app:"vtexhackathon05.iago", termsOfUseAccepted: false) {
    purchased
    billingOptions
  }
}
```
The expected output should be
```
{"data":{"purchase":{"purchased":true,"billingOptions":null}}}
```

When errors occur, the response is always null. E.g., if an app was already purchased for an account:
```
{"data":{"purchase":null},"errors":[{"message":"This app was already purchased","locations":[{"line":2,"column":1}],"path":["purchase"]}]}
```

### CODE REVIEW

File: node/graphql/resolvers/purchase.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { BillingApps, BillingRegistry, Billing, LicenseManager } from '../../clients'
+import { ColossusContext, PurchaseResponse } from '../../entities'
+import { throwAxiosErrorMessage } from '../../handlers'
+import OptionsFormatter from '../../utils/optionsFormatter'
+import { getUserAgent } from '../../utils/conf';
+
+export const purchase = async (_, args: any, ctx: ColossusContext): Promise<PurchaseResponse> => {
+
+    const { appName, termsOfUseAccepted } = args
+    const { vtex: { authToken, account, workspace, region } } = ctx
```
Comment (leticiawanderley) [inline node/graphql/resolvers/purchase.ts]: No purchase não precisa ter workspace, não?

File: node/graphql/resolvers/purchase.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    const ctxOpts = { authToken, userAgent: getUserAgent(), region }
+    const accountOpts = { account, workspace }
+
+    const appsClient = BillingApps.getClient({ ...ctxOpts, ...accountOpts })
```
Comment (leticiawanderley) [inline node/graphql/resolvers/purchase.ts]: Não usado

File: graphql/schema.graphql [hunk 3]
```diff
@@ -1,13 +1,20 @@
 type InstallResponse {
-  installed: Boolean
+  installed: Boolean!
   billingOptions: String
 }
 
+type PurchaseResponse {
+  purchased: Boolean!
+  billingOptions: String
```
Comment (amoreira) [inline graphql/schema.graphql]: the field `billingOptions` it's not necessary, you could return `billingOptions` field and let the client decide the best format to render the response.

File: graphql/schema.graphql [hunk 4]
```diff
@@ -1,8 +1,13 @@
 type InstallResponse {
-  installed: Boolean!
+  code: String
   billingOptions: String
 }
 
+type PurchaseResponse {
+  purchased: Boolean!
+  billingOptionsJSON: String
```
Comment (leticiawanderley) [inline graphql/schema.graphql]: Remover "JSON"

**Pull request merged**